src/solc_0.7/catalyst/AssetUpgrader.sol

```diff
@@ -151,6 +151,8 @@ contract AssetUpgrader is WithMetaTransaction {
     ) internal {
         if (from != to) {
             _asset.safeTransferFrom(from, to, assetId);
+        } else {
+            require(_asset.balanceOf(from, assetId) > 0, "NOT_AUTHORIZED_ASSET_OWNER");
```

Comment:
maybe in the require statement we should add `|| _asset.balanceOf(from) > 0` to handle the case where the asset was minted as an erc721 token (or has been extracted).

Reply:
It seems that _asset.balanceOf supports both

Reply:
yes balanceOf is erc721 method 
and balanceOf(from, assetId) will still return 1 for ERC721 so that's correct as it is.

ERC1155 should have used different naming :(